SurfaceTopography/IO/NC.py

```diff
@@ -175,22 +229,43 @@ def topography(self, channel_index=None, physical_sizes=None,
                    height_scale_factor=None, info={},
                    periodic=None,
                    subdomain_locations=None, nb_subdomain_grid_pts=None):
-        if channel_index is None:
-            channel_index = self._default_channel_index
+        if channel_index is not None and channel_index != 0:
+            raise ValueError('`channel_index` must be None or 0.')
 
-        physical_sizes = self._check_physical_sizes(physical_sizes,
-                                                    self._physical_sizes)
         _info = self._info.copy()
         _info.update(info)
         if subdomain_locations is None and nb_subdomain_grid_pts is None:
-            return Topography(self._heights_var[...], physical_sizes,
-                              periodic=self._periodic
-                              if periodic is None else periodic,
-                              info=_info)
+            if self._x_dim is not None:
+                physical_sizes = self._check_physical_sizes(physical_sizes,
+                                                            self._physical_sizes)
+
+                # This is a uniform topography...
+                if self._y_dim is not None:
+                    # ...and it is 2D
+                    return Topography(np.array(self._heights_var[...]), physical_sizes,
+                                      periodic=self._periodic if periodic is None else periodic,
+                                      info=_info)
+                else:
+                    return UniformLineScan(np.array(self._heights_var[...]), physical_sizes,
+                                           periodic=self._periodic if periodic is None else periodic,
+                                           info=_info)
+            else:
+                if physical_sizes is not None:
+                    raise ValueError('You cannot specify physical sizes for a nonuniform topography.')
+
+                # This is a nonuniform line scan
+                return NonuniformLineScan(np.array(self._x_var[...]), np.array(self._heights_var[...]),
+                                          periodic=self._periodic if periodic is None else periodic,
+                                          info=_info)
         else:
-            return Topography(self._heights_var[...], physical_sizes,
-                              periodic=self._periodic
-                              if periodic is None else periodic,
+            if self._y_dim is None:
+                raise ValueError('Parallel reading only works for topographies, not line scans.')
+
+            physical_sizes = self._check_physical_sizes(physical_sizes,
+                                                        self._physical_sizes)
+
+            return Topography(np.array(self._heights_var[...]), physical_sizes,
```

Comment:
If you call self._heights_var[subdomain_slices], does it only read load the subdomain slice ?

Reply:
Yes. I guess that is implementation specific, but I belive that both `netCDF4` and `scipy.io.netcdf` will just read that subsection. (In `scipy.io.netcdf` it is opened as a memory-mapped buffer anyway.)